script/zmtest

```diff
@@ -0,0 +1,46 @@
+#!/bin/sh
+
+bindir="`dirname $0`"
+
+ZMB="${bindir}/zmb"
+JQ="`which jq`"
+
+error () {
+    status="$1"
+    message="$2"
+    echo "error: ${message}" >&2
+    exit "${status}"
+}
+
+zmb () {
+    json="`"${ZMB}" "$@"`"
+    if echo "${json}" | "${JQ}" -e .error > /dev/null ; then
+        error 1 "method $1: ${json}"
+    else
+        echo "${json}"
+    fi
+}
+
+[ -n "${JQ}" ] || error 2 "Dependency not found: jq"
+
+[ $# -ge 1 ] || error 2 "No domain specified"
+domain="$1" ; shift
+
+# Start test
+testid="`zmb start_domain_test --domain "${domain}" | "${JQ}" -r .result`" || exit 1
+
+# Wait for test to finish
+while true
+do
+    progress="`zmb test_progress --testid "${testid}" | "${JQ}" -r .result`" || exit 1
+    printf "\r${progress}%% done" >&2
+    if [ "${progress}" -eq 100 ] ; then
+        echo >&2
+        break
+    fi
+    sleep 1
+done
+
+# Get test results
+zmb get_test_results --testid "${testid}" --lang en
+echo "testid:" ${testid} >&2
```

Comment:
Add an extra "echo" before to separate it from the JSON blob.

Reply:
In my view it's better the way it is. It is clearly visible to the human consumer that the testid line is not part of the JSON blob, and it's also clearly visible to the machine consumer that the two parts are separated into different streams.